File: CustomerViewDemo/app/src/main/java/com/example/customerviewdemo/MainFragment.java
package com.example.customerviewdemo;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.ArrayList;

/**
 * Created by Administrator on 2017/6/28.
 */

public class MainFragment extends Fragment {

    RecyclerView rv_main;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view=inflater.inflate(R.layout.fragment_main, container, false);
        rv_main=view.findViewById(R.id.rv_main);
        rv_main.setHasFixedSize(true);
        rv_main.setLayoutManager(new LinearLayoutManager(getActivity()));
        ArrayList<String> strings=new ArrayList<>();
        strings.add("");
        strings.add("");
        strings.add("");
        strings.add("");
        strings.add("");
        strings.add("");
        strings.add("");
        strings.add("");
        strings.add("");
        strings.add("");
        strings.add("");
        strings.add("");
        MainAdapter adapter=new MainAdapter(strings, getActivity());
        rv_main.setAdapter(adapter);
        return view;
    }

    public RecyclerView getRv_main() {
        return rv_main;
    }
}

File: GyroscopeImageDemo/app/src/main/java/com/example/administrator/gyroscopeimagedemo/ui/main/MainFragment.kt
package com.example.administrator.gyroscopeimagedemo.ui.main

import android.arch.lifecycle.ViewModelProviders
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.example.administrator.gyroscopeimagedemo.R
import com.example.administrator.gyroscopeimagedemo.ui.adapter.MainAdapter
import com.example.administrator.gyroscopeimagedemo.utils.GyroscopeObserver
import com.example.administrator.gyroscopeimagedemo.utils.GyroscopeTransFormation
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.main_fragment.*


class MainFragment : Fragment() {

    val images: ArrayList<String> by lazy {
        ArrayList<String>()
    }

    companion object {
        fun newInstance() = MainFragment()
    }

    private lateinit var viewModel: MainViewModel

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View {
        return inflater.inflate(R.layout.main_fragment, container, false)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        viewModel = ViewModelProviders.of(this).get(MainViewModel::class.java)

        Picasso.get()
                .load("http://7b1g8u.com1.z0.glb.clouddn.com/house1.png")
                .transform(GyroscopeTransFormation(1080, 600))
                .into(iv_image)

//        Glide.with(iv_image.context)
//                .load("http://7b1g8u.com1.z0.glb.clouddn.com/house1.png")
//                .into(iv_image)

        images.add("http://7b1g8u.com1.z0.glb.clouddn.com/house4.png")
        images.add("http://7b1g8u.com1.z0.glb.clouddn.com/house1.png")
        images.add("http://7b1g8u.com1.z0.glb.clouddn.com/house2.png")
        images.add("http://7b1g8u.com1.z0.glb.clouddn.com/house3.png")
        images.add("http://7b1g8u.com1.z0.glb.clouddn.com/house4.png")
        images.add("http://7b1g8u.com1.z0.glb.clouddn.com/house1.png")

        rv_image.setHasFixedSize(true)
        rv_image.scrollToPosition(1)
        rv_image.addOnItemChangedListener { _, adapterPosition ->
            if (adapterPosition == 0) {
                rv_image.scrollToPosition(4)
            }
            if (adapterPosition == 5) {
                rv_image.scrollToPosition(1)
            }
        }
        val adapter = MainAdapter(requireContext(), images)
        rv_image.adapter = adapter
    }

    override fun onResume() {
        super.onResume()
        GyroscopeObserver.getInstance().register(requireContext())
    }

    override fun onPause() {
        super.onPause()
        GyroscopeObserver.getInstance().unRegister()
    }
}
